### PR TITLE
localtime: use upstream unit, fix polkit rules

### LINKED ISSUE #55288: localtimed user can't set timezone
## Issue description

I've just set `services.localtime.enable = true` on my laptop while on holiday. The timezone never updates (tried restarting, waiting a while).

Running `journalctl -b -u localtime` I see:

```
-- Logs begin at Thu 2018-12-27 01:52:52 NZDT, end at Wed 2019-02-06 12:17:39 NZDT. --
Feb 06 12:02:39 sima systemd[1]: Started localtime service.
Feb 06 12:03:02 sima localtimed[804]: Failed to set timezone: Pacific/Auckland
Feb 06 12:03:03 sima localtimed[804]: Failed to set timezone: Pacific/Auckland
```

I see in https://github.com/NixOS/nixpkgs/blob/release-18.09/nixos/modules/services/system/localtime.nix that the `localtimed` service is run as the `localtimed` user.

If I run: `sudo -u localtimed localtimed` I also see the message `Failed to set timezone: Pacific/Auckland`.

Whereas if I run `sudo localtimed` there is no output and the timezone immediately updates (as shown by an immediate change in the time in my status bar, and confirmed by running `timedatectl`).

It seems that the `localtimed` user just doesn't have permission to set the timezone.

### Steps to reproduce

1. Set `services.localtime.enable = true` in `/etc/nixos/configuration.nix` (and unset `time.timeZone`)
2. `sudo nixos-rebuild switch`
3. Wait to see if timezone changes
4. `journalctl -b -u localtime`, look for messages with "Failed to set timezone" (will only appear if the timezone it would need to set is different than the current timezone)

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.16, NixOS, 19.03pre166987.bc41317e243 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `"nixos-19.03pre166987.bc41317e243"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### PR DESCRIPTION
**Pull request by michaelpj**
###### Motivation for this change
Fixes #55288.

`localtime`'s systemd unit and polkit rules weren't being installed since ed473e661533d04d3ed212be837346430c73cc88, which moves it to using `buildGoPackage`. This caused two issues:
- We weren't using `make install`, so we weren't installing all the build products to the output at all.
- `buildGoPackage` puts the `bin` output first, so the bare package references in the module were getting the `bin` output rather than the `out` output, and so wouldn't see the polkit rules anyway.

This PR fixes those issue.s

Along the way, I made us actually use the upstream unit, which works fine in my testing. With these changes and the recent geoclue changes, localtime now actually works on my system.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by michaelpj** [comment]
cc @Mic92 perhaps, who merged https://github.com/NixOS/nixpkgs/pull/52627?

File: pkgs/tools/system/localtime/default.nix [hunk 1]
```diff
@@ -11,18 +11,17 @@ buildGoPackage rec {
   };
   goPackagePath = "github.com/Stebalien/localtime";
 
-  buildInputs = [ systemd polkit m4 ];
+  buildInputs = [ m4 ];
 
-  makeFlags = [ "PREFIX=$(out)" ];
+  makeFlags = [ "PREFIX=$(out)" "BINDIR=$(bin)/bin" ];
```
Comment (worldofpeace) [inline pkgs/tools/system/localtime/default.nix]:
```suggestion
  makeFlags = [ 
    "PREFIX=${placeholder ''out''}"
    "BINDIR=${placeholder ''out''}/bin"
  ];
```

**Comment by worldofpeace** [comment]
I've tested that at least the services would start in a vm.

File: pkgs/tools/system/localtime/default.nix (hunk 1 above)
Comment (michaelpj) [reply pkgs/tools/system/localtime/default.nix]: Done, thanks.

**Comment by worldofpeace** [comment]
Thanks for fixing this @michaelpj, it's a great improvement.

**Pull request merged**